components/automate-cli/pkg/verifyserver/services/systemresourceservice/systemresourcesservice.go

```diff
@@ -0,0 +1,341 @@
+package systemresourceservice
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/logger"
+)
+
+type ISystemResourcesService interface {
+	GetSystemResourcesForDeployment(string, string) (*models.ApiResult, error)
+}
+
+type SystemResourcesService struct {
+	logger                 logger.Logger
+	GetOsAndFileSystemInfo IGetOsAndFileSystemInfo
+}
+
+func NewSystemResourceService(log logger.Logger, osFsUtil IGetOsAndFileSystemInfo) *SystemResourcesService {
+	return &SystemResourcesService{
+		logger:                 log,
+		GetOsAndFileSystemInfo: osFsUtil,
+	}
+}
+
+const (
+	CPU_INFO_FILE = "/proc/cpuinfo"
+
+	CPU_COUNT_CHECK_TITLE   = "CPU count check"
+	CPU_SPEED_CHECK_TITLE   = "CPU speed check"
+	MEMORY_SIZE_CHECK_TITLE = "Memory size check"
+	FREE_SPACE_CHECK        = "%s free space check"
+
+	SUCCESS_MSG        = "%s should have free space >=%vGB"
+	ERROR_MSG          = "%s free space is %0.2fGB"
+	SUCCESS_MSG_IN_PER = " or %v%% of total size of /hab"
+
+	INVALID_NODE_TYPE_ERR = "given query node_type with value=%s is not supported"
+	RESOLUTION_MSG        = "Please run system on supported platform"
+)
+
+func (srs *SystemResourcesService) GetSystemResourcesForDeployment(nodeType, deploymentState string) (*models.ApiResult, error) {
+
+	srsResponse := &models.ApiResult{
+		Passed: true,
+		Checks: []models.Checks{},
+	}
+
+	cpuCountCheck := srs.GetCpuCountCheck()
+	if !cpuCountCheck.Passed {
+		srsResponse.Passed = false
+	}
+	srsResponse.Checks = append(srsResponse.Checks, *cpuCountCheck)
+
+	cpuSpeedCheck := srs.GetCpuSpeedCheck(CPU_INFO_FILE)
+	if !cpuSpeedCheck.Passed {
+		srsResponse.Passed = false
+	}
+	srsResponse.Checks = append(srsResponse.Checks, *cpuSpeedCheck)
+
+	memorySizeCheck := srs.GetMemorySizeCheck()
+	if !memorySizeCheck.Passed {
+		srsResponse.Passed = false
+	}
+	srsResponse.Checks = append(srsResponse.Checks, *memorySizeCheck)
+
+	habFreeSpaceCheck, err := srs.GetFreeHabSpace(nodeType, deploymentState)
+	if err != nil {
+		return nil, err
+	}
+
+	if !habFreeSpaceCheck.Passed {
+		srsResponse.Passed = false
+	}
+	srsResponse.Checks = append(srsResponse.Checks, *habFreeSpaceCheck)
+
+	tmpFreeSpaceCheck := srs.GetFreeDiskSpaceCheckOfDir("/tmp", constants.TMP_FREE_DISK_IN_PER, constants.TMP_FREE_DISK_IN_GB, "Temp")
```

Comment:
`GetFreeDiskSpaceCheckOfDir` function name doesn’t make any sense what it doing.

Reply:
Done